pkg/framework/machines.go

```diff
@@ -103,28 +103,34 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
 	return machine, nil
 }
 
-// DeleteMachine deletes a specific machine and returns an error otherwise
-func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) error {
+// DeleteMachines deletes a specific machines and returns an error otherwise
```

Comment:
just a nit about the comment,

```suggestion
// DeleteMachines deletes the specified machines and returns an error on failure.
```